flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -265,6 +272,29 @@ protected void initialize() throws ResourceManagerException {
 		}
 	}
 
+	@Override
+	public CompletableFuture<RegistrationResponse> registerTaskExecutor(
```

Comment:
I think it's better to override `workerStarted()`

```
	@Override
	protected YarnWorkerNode workerStarted(ResourceID resourceID) {
		// ack the pending register
		pendingContainersExpectedToRegister.remove(resourceID);
		return workerNodeMap.get(resourceID);
	}
```

This method is called when the registration was successful. When `registerTaskExecutor` is called, the registration can still fail.

Reply:
Good point! Have to change.